src/test_lib_json/main.cpp

```diff
@@ -3724,7 +3724,7 @@ JSONTEST_FIXTURE_LOCAL(IteratorTest, reverseIterator) {
   using Iter = decltype(json.begin());
   auto re = std::reverse_iterator<Iter>(json.begin());
   for (auto it = std::reverse_iterator<Iter>(json.end()); it != re; ++it) {
-    values.push_back(it->asString());
+    values.push_back((*it).asString());
```

Comment:
`it->asString` should work without being changed. Revert this line if you can.

Reply:
It should work, but it doesn't. I looked through all the documentation for reverse_iterator, and I can find no reason why "->" fails.
My next step would be to check clang to see if the same error appears there

Reply:
Friendly ping, I definitely think the iterator definition is a concern.

On my macbook with clang 12 I don't see this compiler error, it's definitely limited to GCC. I'll try to build on GCC when I have a chance and see what's going on.

Reply:
Figured this out and pushed a fix for it to this branch.

https://github.com/open-source-parsers/jsoncpp/pull/1216#discussion_r519011218